### DISCUSSION TITLE
demultiplexer pattern?

### DISCUSSION THREAD

**Discussion by cong-or** [tokio-rs/tracing#1961] [Q&A] [answered]
is it possible to collect logs and distribute the data to different sinks? For example, pipe data to `stdout` , `text file` and **`custom endpoint`**? 

** in regards to custom endpoint, is it possible to write custom logic to send over the wire etc? **

**Comment by davidbarsky** [comment, marked as answer]
> is it possible to collect logs and distribute the data to different sinks? For example, pipe data to stdout , text file and custom endpoint?

Yes. I'd look at this example: https://github.com/tokio-rs/tracing/blob/v0.1.x/examples/examples/fmt-multiple-writers.rs.

> ** in regards to custom endpoint, is it possible to write custom logic to send over the wire etc? **

Yes. The common approach is to implement your own [Layer](https://docs.rs/tracing-subscriber/0.3.9/tracing_subscriber/layer/trait.Layer.html). The layers in tracing-subscriber are somewhat complex to learn from, but the formatting logic inside of https://github.com/davidbarsky/tracing-tree aside, it could be a good starting point to understand what a simpler layer can look like.

**Reply by cong-or** [reply]
thanks!
